akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2Blueprint.scala

```diff
@@ -46,23 +47,35 @@ private[http2] sealed trait Http2SubStream[+T] {
       /* contentLength undefined */
       case c: ChunkedHttp2SubStream => HttpEntity.Chunked(contentType, c.data)
     }
+
+  def createResponseEntity(contentLength: Long, contentType: ContentType): RequestEntity
 }
 
 private[http2] final case class ByteHttp2SubStream(
   initialHeaders:        ParsedHeadersFrame,
   data:                  Source[ByteString, Any],
   correlationAttributes: Map[AttributeKey[_], _] = Map.empty
-) extends Http2SubStream[ByteString] {
-  override def withCorrelationAttributes(attributes: Map[AttributeKey[_], _]): Http2SubStream[ByteString] =
+) extends Http2SubStream {
+  type T = ByteString
+  override def withCorrelationAttributes(attributes: Map[AttributeKey[_], _]): Http2SubStream =
     copy(correlationAttributes = attributes)
+
+  def createResponseEntity(contentLength: Long, contentType: ContentType): RequestEntity = {
```

Comment:
This is the only thing I had to add to make the simpler types work. I'm not too concerned about having to add it though.